src/dash-table/type/reconcile.ts

```diff
@@ -63,16 +63,14 @@ function doFailureRecovery(result: IReconciliation, c: IColumnType) {
     const failure = (c && c.on_change && c.on_change.failure) || ChangeFailure.Reject;
     result.failure = failure;
 
-    // If Skip/Prevent
-    if (failure !== ChangeFailure.Default) {
-        return result;
+    if (failure === ChangeFailure.Default) {
+        const defaultValue = (c && c.validation && c.validation.default) || null;
+        result.success = true;
+        result.value = defaultValue;
+    } else if (failure === ChangeFailure.Accept) {
+        result.success = true;
```

Comment:
The logic here was incorrect and based off the previous Skip/Prevent implementation -- the new tests uncovered the case as it was not properly tested.